packages/generate-persisted-query-manifest/src/index.ts

```diff
@@ -482,13 +500,20 @@ export async function generatePersistedQueryManifest(
       continue;
     }
 
+    let documentCount = 0;
+
     // We delegate validation to the functions that return the document sources.
     // We just need to record the operations here to sort them in the manifest
     // output.
     visit(source.node, {
       OperationDefinition(node) {
         const name = node.name?.value;
 
+        if (++documentCount > 1) {
+          addError(source, ERROR_MESSAGES.multipleOperations());
+          return BREAK;
+        }
+
         if (!name) {
```

Comment:
Hmm, I'm wondering why we don't have any sort of error or warning on anonymous operations rather than just not including them in the manifest and having them break...

Reply:
Hmmm, I'm not sure I follow what you're asking (or I'm reading your comment wrong). Currently we error on anonymous operations and force you to name all of them. Here is a test that shows this behavior: https://github.com/apollographql/apollo-utils/blob/930dee9b49e0d143e80aa610c68474ff2fed453e/packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts#L1255-L1277

Are you seeing something different in testing?

Reply:
Finally realized what I was missing which is that this check was done in a different code block that traversed the AST in `fromFilepathList`. For now I moved the check for the number of operations into that same codeblock. It looks like GraphQL Codegen will split those operations into separate entries in the manifest.

If we decide to include a fail-safe and would prefer we don't allow this at all, we can move the check back and update the comment explaining why we do this check in the original place rather than moving it to the function that returns the document sources.